packages/amplify_datastore/example/integration_test/save_test.dart

```diff
@@ -62,5 +62,21 @@ void main() {
         expect(updatedBlogs[0].name, updatedBlogName);
       },
     );
+
+    testWidgets('predicate should prevent save to non matching model',
```

Comment:
*nit:* Same as above. Might be nice to add a similar test where the model doesn't match the predicate.